includes/postratings-comment.php

```diff
@@ -6,8 +6,12 @@
 
 add_filter( 'wp_insert_comment', 'process_ratings_from_comment' );
 function process_ratings_from_comment($comment_id) {
-		$post_id = intval($_POST['comment_post_ID']);
-    $rate = intval($_POST["wp_postrating_form_value_$post_id"]);
+    if ( !isset($_POST['comment_post_ID']) || !isset($_POST['wp_postrating_form_value_' . $post_id]) ) {
+        return;
+    }
+
+    $post_id = intval($_POST['comment_post_ID']);
```

Comment:
```
$post_id = isset( $_POST['comment_post_ID'] ) ? (int) $_POST['comment_post_ID'] : 0;
$rate = isset( $_POST['wp_postrating_form_value_' . $post_id]; ) ? (int) $_POST['wp_postrating_form_value_' . $post_id ] : 0;

if( 0 === $post_id  || 0 === $rate ) {
    return;
}
```

I might be nitpicking and double standard, but going forward, I will be using (int) than intval() as (int) is much faster. I recommend coding using PHPStorm and https://plugins.jetbrains.com/plugin/7622-php-inspections-ea-extended-

Been using it for my workplace code and it is amazing.

Reply:
`intval(q)` => `(int)` : done
About strict type checking I'm not totally fond of it here because not-set/empty/FALSE/0 are semantically identical in this case and I'd prefer to globally catch and reject any user-provided edge-case.
I've also the feeling that giving a default value (0) that we're going to reject a couple of lines after makes code understanding a bit more difficult.

NB: Emacs is a nice code-editor BTW